CONTRIBUTE.adoc

```diff
@@ -0,0 +1,104 @@
+_contribution_ guidelines layed out in this document.
+
+toc::[]
+
+== Writing
+
+. The guidelines are written in https://asciidoctor.org[asciidoc as described
+  by Asciidoctor].
+. each guideline is made of one sentence, as easy to remember as possible,
+  followed by a collapsible description, made of:
+** explanations
+** rationale
+** examples
++
+The result looks then as the following template shows (you may copy & paster):
++
+[source,asciidoc]
+------------------------------------------------------------------------
+.[big]#**Do this and do not do that is the guideline**#
+[%collapsible]
+====
+Explanations:: These are explanations
+
+Rationale:: This is the rationale
+
+Examples:: These are examples
++
+[source,yaml]
+----
+- name: a mini example of playbook
+  hosts: all
+  gather_facts: false
+  become: false
+
+  tasks:
+
+  - name: say what we all think
+    debug:
+      msg: asciidoctor is {{ my_private_thoughts }}
+----
++
+Even more examples...
+====
+------------------------------------------------------------------------
++
+. Those guidelines are grouped into sections and optionally sub-sections,
+  as far as required for maintainability.
+. Those (sub-)sections can be written in their own source file, but then
+  are included with `include::directory/file.adoc[leveloffset=1]` in the
+  parent section's file, so that all source files are interlinked and can
+  be rendered all together by rendering the top `README.adoc`, either with
+  `asciidoctor` or with `asciidoctor-pdf`.
++
+NOTE: this contribution file is obviously not meant for inclusion in the
+      overall document.
++
+. Each source file has a single title (the line starting with one equal sign)
+  and can be rendered individually (the `leveloffset` is set such that it
+  fits in the overall headings structure when included).
+. The source code is written as readable as possible in its raw form.
+  Lines are kept below 80 characters, indentations are respected.
+. Sentences are written in the present tense form, avoid "should", "must", etc
+  For example, "Sentences are written", not "Sentences should be written" or
+  "Sentences must be written". This avoids filler words.
+. The https://en.wikipedia.org/wiki/Singular_they[singular "they"] is used to
+  avoid the unreadable "he/she/it" construct and still be neutral.
+
+== Contributing
+
+. Just fork the repository, create a Pull Request (PR) and offer your changes.
+
+== Example
+
+This is how one guideline as shown above looks like:
+
+.[big]#**Do this and do not do that is the guideline**#
+[%collapsible]
+====
+Explanations:: These are explanations
+
+Rationale:: This is the rationale
+
+Examples:: These are examples
++
+[source,yaml]
+----
+- name: a mini example of playbook
+  hosts: all
+  gather_facts: false
+  become: false
+
+  tasks:
+
+  - name: say what we all think
+    debug:
+      msg: asciidoctor is {{ my_private_thoughts }}
+----
++
+Even more examples...
```

Comment:
This seems to be identical to the template above. Does it really need to be in here twice?

Reply:
It's because you read the raw code, the idea is to have it once as raw code, once rendered as example, so that people see how it looks. It makes more sense if you look at the rendered page. I've added comments and links accordingly to make this clearer.